lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -137,6 +156,75 @@ void SPIRVRegularizeLLVM::lowerMemset(MemSetInst *MSI) {
   return;
 }
 
+void SPIRVRegularizeLLVM::buildFunnelShiftLeftFunc(Function *FSHLFunc) {
+  if (!FSHLFunc->empty())
+    return;
+
+  auto *IntTy = dyn_cast<IntegerType>(FSHLFunc->getReturnType());
+  assert(IntTy && "llvm.fshl: expected an integer return type");
```

Comment:
```suggestion
  auto *IntTy = cast<IntegerType>(FSHLFunc->getReturnType());
```

`cast` already includes an assertion within it.